src/main/views/MattermostView.ts

```diff
@@ -390,61 +367,148 @@ export class MattermostView extends EventEmitter {
+        } catch (err: any) {
+            this.log.error('There was an error trying to request the unread state', err);
         }
     }
 
-    titleParser = /(\((\d+)\) )?(\* )?/g
-
-    handleTitleUpdate = (e: Event, title: string) => {
-        log.debug('handleTitleUpdate', {tabName: this.tab.name, title});
+    private handleTitleUpdate = (e: Event, title: string) => {
+        this.log.debug('handleTitleUpdate', title);
 
         this.updateMentionsFromTitle(title);
     }
 
-    updateMentionsFromTitle = (title: string) => {
-        const resultsIterator = title.matchAll(this.titleParser);
-        const results = resultsIterator.next(); // we are only interested in the first set
-        const mentions = (results && results.value && parseInt(results.value[MENTIONS_GROUP], 10)) || 0;
-
-        appState.updateMentions(this.tab.name, mentions);
-    }
-
-    handleFaviconUpdate = (e: Event, favicons: string[]) => {
-        log.silly('handleFaviconUpdate', {tabName: this.tab.name, favicons});
+    private handleFaviconUpdate = (e: Event, favicons: string[]) => {
+        this.log.silly('handleFaviconUpdate', favicons);
 
         // if unread state is stored for that favicon, retrieve value.
         // if not, get related info from preload and store it for future changes
-        this.currentFavicon = favicons[0];
         this.findUnreadState(favicons[0]);
     }
 
-    // if favicon is null, it will affect appState, but won't be memoized
-    findUnreadState = (favicon: string | null) => {
-        try {
-            this.view.webContents.send(IS_UNREAD, favicon, this.tab.name);
-        } catch (err: any) {
-            log.error(`There was an error trying to request the unread state: ${err}`);
-            log.error(err.stack);
+    /**
+     * Loading/retry logic
+     */
+
+    private retry = (loadURL: string) => {
+        return () => {
+            // window was closed while retrying
+            if (!this.view || !this.view.webContents) {
+                return;
+            }
+            const loading = this.view.webContents.loadURL(loadURL, {userAgent: composeUserAgent()});
+            loading.then(this.loadSuccess(loadURL)).catch((err) => {
+                if (this.maxRetries-- > 0) {
+                    this.loadRetry(loadURL, err);
+                } else {
+                    WindowManager.sendToRenderer(LOAD_FAILED, this.name, err.toString(), loadURL.toString());
+                    this.emit(LOAD_FAILED, this.name, err.toString(), loadURL.toString());
+                    this.log.info(`Couldn't establish a connection with ${loadURL}, will continue to retry in the background`, err);
+                    this.status = Status.ERROR;
+                    this.retryLoad = setTimeout(this.retryInBackground(loadURL), RELOAD_INTERVAL);
+                }
+            });
+        };
+    }
+
+    private retryInBackground = (loadURL: string) => {
+        return () => {
```

Comment:
Similarly, I feel like it'd be a better pattern to have the arrow function wrapper added where we're actually using these. It reminds me of how we used to write Go functions that immediately started a goroutine instead of making them async and wrapping the function call in a goroutine. It made testing them absolutely miserable.